src/tests/ftest/util/osa_utils.py

```diff
@@ -487,7 +487,7 @@ def run_ior_thread(self, action, oclass, test, single_cont_read=True,
         process.start()
         # Wait for the thread to finish
         process.join()
-        if not self.out_queue.empty():
+        if fail_on_warning is True and not self.out_queue.empty():
```

Comment:
It's not necessary to use `True` explicitly
```suggestion
        if fail_on_warning and not self.out_queue.empty():
```

Reply:
Done.